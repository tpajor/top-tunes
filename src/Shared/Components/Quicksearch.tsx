import * as React from 'react';
import { Subject } from 'rxjs';

interface IProps {
  term$: Subject<string>;
}

class Quicksearch extends React.PureComponent<IProps> {
  private readonly handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    this.props.term$.next(event.target.value);
  }

  public render() {
    return (
      <div className='container container--form-control'>
        <input
          type='text'
          className='input input--search'
          placeholder='Type here to filter the list'
          onChange={this.handleInputChange}
        />
        <div className='spyglass' />
      </div>
    );
  }
}

export default Quicksearch;
