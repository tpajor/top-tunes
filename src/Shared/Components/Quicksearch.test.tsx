import * as React from 'react';
import { shallow } from 'enzyme';
import { Subject } from 'rxjs';

import Quicksearch, { IProps } from './Quicksearch';

function setup() {
  const props: IProps = {
    term$: new Subject<string>(),
  };

  const wrapper = shallow(<Quicksearch {...props} />);

  return {
    wrapper,
    props,
  };
}

describe('Quicksearch component', () => {
  it('should match snapshot if no change was intended', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('Shuld add term to Subject on input value change', () => {
    const { wrapper } = setup();
    const event = {
      target: {
        value: 'test',
      },
    };
    spyOn(Subject.prototype, 'next');

    wrapper.find('input').simulate('change', event);

    expect(Subject.prototype.next).toHaveBeenCalledWith(event.target.value);
  });
});
