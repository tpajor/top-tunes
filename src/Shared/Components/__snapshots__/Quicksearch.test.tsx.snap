// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quicksearch component should match snapshot if no change was intended 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Quicksearch
    term$={
      Subject {
        "_isScalar": false,
        "closed": false,
        "hasError": false,
        "isStopped": false,
        "observers": Array [],
        "thrownError": null,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="input input--search"
          onChange={[Function]}
          placeholder="Type here to filter the list"
          type="text"
        />,
        <div
          className="spyglass"
        />,
      ],
      "className": "container container--form-control",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input input--search",
          "onChange": [Function],
          "placeholder": "Type here to filter the list",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "spyglass",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="input input--search"
            onChange={[Function]}
            placeholder="Type here to filter the list"
            type="text"
          />,
          <div
            className="spyglass"
          />,
        ],
        "className": "container container--form-control",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input input--search",
            "onChange": [Function],
            "placeholder": "Type here to filter the list",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "spyglass",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
